packages/core/src/docdb/commands/modifyInstance.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { DBClusterNode } from '../explorer/dbClusterNode'
+import { DBInstanceNode } from '../explorer/dbInstanceNode'
+import { DBCluster, ModifyDBInstanceMessage } from '@aws-sdk/client-docdb'
+import { DBStorageType, DocumentDBClient } from '../../shared/clients/docdbClient'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { isValidResponse } from '../../shared/wizards/wizard'
+
+/**
+ * Modifies a DocumentDB instance.
+ *
+ * Prompts the user for the instance class.
+ * Updates the instance.
+ * Refreshes the node.
+ */
+export async function modifyInstance(node: DBInstanceNode) {
+    getLogger().debug('ModifyInstance called for: %O', node)
+
+    if (!node) {
+        throw new Error('No node specified for ModifyInstance')
+    }
+
+    if (node.instance.DBInstanceStatus !== 'available') {
+        void vscode.window.showErrorMessage(
+            localize('AWS.docdb.modifyInstance.instanceStopped', 'Instance must be running')
+        )
+        return
+    }
+
+    const instanceName = node.instance.DBInstanceIdentifier
+    const parent = node.parent as DBClusterNode
+
+    const quickPickItems = await getInstanceClassOptions(parent.client, parent.cluster)
+    const newInstanceClass = await getInstanceClass(quickPickItems, node.instance.DBInstanceClass!)
```

Comment:
**Nit**

It wasn't immediately clear that this is performing a UI step. Perhaps a different name like `promptForInstanceClass()` might make it more clear.